### ISSUE TITLE
feat: Support Message Streaming (like ChatGPT)

### ISSUE THREAD

**Issue by DelattreAdixon** [chatwoot/chatwoot#10315]
### Is your feature or enhancement related to a problem? Please describe.

In a scenario where an AI LLM is integrated as an Agent Bot in Chatwoot, it offers a great advantage: the human user can engage in a complete conversation with the AI LLM (such as ChatGPT, LLaMA, or any other model). Moreover, the Agent Bot is relatively straightforward to develop.

However, for the Agent Bot to send a message via the Chatwoot API, the full message must be available. Since LLMs may take a few seconds to generate text, this poses a challenge. This is why ChatGPT (and other LLMs) utilize message streaming, where words are displayed progressively to the human user as the model continues to generate the rest of the response. This way, the user doesn't have to wait for the entire message to be computed before seeing the response.

### Describe the solution you'd like

It would be highly beneficial if the API could support message streaming. This would allow the message to be displayed progressively on the human's side, similar to how it works in ChatGPT.

### Describe alternatives you've considered

An alternative would be to split the LLM message into multiple messages. For instance, when the LLM finish a sentence, send this sentence to the Chatwoot API.

**Comment by nyamkamunhjin** [comment]
Any update on this?

**Comment by maxvornovskykh** [comment]
It would solve a lot of pain points 🤞

**Comment by vitaliierfan1** [comment]
This is a good one, hope to see it added soon!

**Comment by Anastasiia-Rem** [comment]
Really eager to see it rolled out!

**Comment by Bobbysimicua** [comment]
I need it 🥇

**Comment by blazzy1** [comment]
This feature is much needed! Excited to see where this goes

**Comment by ayefimenko** [comment]
Would love to see it come to life

**Comment by YaroslavKryz** [comment]
This would be a fantastic addition! Really hope it gets implemented soon

**Comment by VictorVas07** [comment]
This would be a real game-changer for Chatwoot + AI integrations. Streaming output keeps users engaged while the model is still generating a response, instead of leaving them waiting in silence. Really hope to see this implemented soon!

**Comment by semernya** [comment]
Great idea 💡

**Comment by OksanaPid** [comment]
Completely support this request.
The lack of message streaming support is actually the main blocker for me and my team when considering Chatwoot as our primary tool. Real-time, progressive message delivery is essential for a smooth user experience when integrating LLMs, and without it, the interaction feels noticeably less natural.

**Comment by DanielVasovsky** [comment]
Awesome feature, hope to see it added!

**Comment by HelgaUa** [comment]
Absolutely, that’s an excellent idea!

**Comment by nm-369** [comment]
This feature is a game-changer! Hyped to see how it evolves!

**Comment by Uliana105** [comment]
Definitely a great idea!

**Comment by maxpol888** [comment]
This functionality would be useful in many use cases!

**Comment by TayaFish** [comment]
Looking forward!

**Comment by LightNeO** [comment]
Wow, so will it be implemented soon?

**Comment by EgorKrupyna** [comment]
Just one question, "WHEN?"
Excited to see 🤩

**Comment by maxvornovskykh** [comment]
Looks like n8n [quietly added ](https://community.n8n.io/t/stream-ai-responses-on-http-responses-llm-chains-and-ai-agents-nodes/52084/31)response streaming support in 1.105.2. Your turn to support response streaming in chatwoot

**Comment by ye1izaveta** [comment]
Hope to see it added soon!

**Comment by MykhailoHordun** [comment]
Hope to see it added soon!!!

**Comment by oleksandra-synytska** [comment]
I agree, this feature would really enhance the user experience

**Comment by fAtsteD** [comment]
It will be helpful to have an answer continuously without immediate, so it would look like the answer is mostly instant to the user.
Some LLMs have long requests for full response, but with streaming, it does not look so.

**Comment by ViktoriiaPoliakova110448** [comment]
I think this could make a big difference, this feature would really enhance the user experience

**Comment by LiliDov** [comment]
Really hope this gets prioritized — it would add so much value and make the experience far smoother

**Comment by oksana-oksana-d** [comment]
This feature would be a real game-changer! I’d love to see it in action, as it could significantly make users’ lives easier and make the service even more convenient.

**Comment by olhatkachenko-web** [comment]
Would love to see it come to life!

**Comment by anton9qa-ui** [comment]
Can’t wait for it to be added

**Comment by Hollyday2089** [comment]
Sounds great

**Comment by OleksaHaman** [comment]
Yep, yep, yep, that would be great. Fingers crossed to have implementation)

**Comment by popovich668** [comment]
Finally! We are waiting

**Comment by darynaskr** [comment]
Excited to see this get prioritized!

**Comment by kinzer-m** [comment]
Can we help smhow to speed this up? ⏩ ⏩ ⏩

**Comment by JonasReuter** [comment]
I’ve opened a PR that adds minimal server-side “streaming” for webhook bots and fixes the AgentBot sender serialization issue.

PR: #12396

## What it does:

Adds start/append/finish endpoints to progressively update the bot’s outgoing message and broadcast message.updated over ActionCable. start/finish also toggle typing_on/off.
Whitelists these actions for AgentBot tokens.
Fixes NoMethodError by adding AgentBot#pubsub_token (returns empty string), so serializers/broadcasters don’t crash when the sender is AgentBot.

## Outcome:

Bot replies can be streamed to the UI without HTTP chunked responses.
No more 422/NoMethodError when AgentBot sends a message.
